tracing/src/main/java/com/palantir/tracing/TraceLocalObserver.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tracing;
+
+public interface TraceLocalObserver<T> {
+    void onTraceComplete(T value);
```

Comment:
Should this include other attributes from the `RequestState` e.g. `traceId`, `optional<requestI>`, etc? Such values could also be included in the provided threadlocal value if necessary, but I suspect the traceId at minimum would be helpful for reporting (allows us to tie anything that we report back to the larger operation)

Reply:
yeah makes sense, will add that in the second PR adding observers